node/clients/intelligent-search-api.ts

```diff
@@ -0,0 +1,114 @@
+import { ExternalClient, InstanceOptions, IOContext } from "@vtex/api";
+import { parseState } from "../utils/searchState";
+
+interface CorrectionParams {
+  query: string
+}
+
+interface SearchSuggestionsParams {
+  query: string
+}
+
+interface AutocompleteSearchSuggestionsParams {
+  query: string
+}
+
+interface BannersArgs {
+  query: string
+}
+
+interface FacetsArgs {
+  query?: string
+  page?: number
+  count?: number
+  sort?: string
+  operator?: string
+  fuzzy?: string
+  leap?: boolean
+  tradePolicy?: number
+  searchState?: string
+  hideUnavailableItems?: boolean | null
+  removeHiddenFacets?: boolean | null
+  options?: Options
+  initialAttributes?: string
+  workspaceSearchParams?: object
+  regionId?: string | null
+}
+
+const decodeQuery = (query: string) => {
+  try {
+    return decodeURIComponent(query)
+  } catch (e) {
+    return query
+  }
+}
+
+export class IntelligentSearchApi extends ExternalClient {
+  private locale: string | undefined
+  // private store: string
+
+  public constructor(context: IOContext, options?: InstanceOptions) {
+    super(`http://${context.workspace}--${context.account}.myvtex.com/_v/api/intelligent-search`, context, {
```

Comment:
shouldn't we use always the production workspace?

Reply:
If we always use the production workspace will it be possible to debug a linked version?
I mean, I am on workspace `hiago` and I link a dev version of the `vtex.intelligent-search-api`, I would like to have the dev version called, and not the production one.

Reply:
In this case we could use `AppClient` instead of `ExternalClient`, since that already does it. What do you think?

Reply:
I tried before, but I was not able to make the `AppClient` work at all. [AppClient always call the app with the `http://${service}.${region}.vtex.io/${account}/${workspace}` base URL](https://github.com/vtex/node-vtex-api/blob/08ea11d380997f5abf02455487b342caa74b2001/src/clients/apps/AppClient.ts#L24), but this URL pattern only works for private endpoints and the [endpoints on `vtex.intelligent-search-api` are public)](https://github.com/vtex/intelligent-search-api/blob/main/node/service.json#L23).

I'm not sure if I'm missing something, but I looked at one example of public endpoints and they also used ExternalClient. Should we change the endpoints to private? Or there is any other way?